### PR TITLE
Importing remote sources over HTTP(S)

### PR DESCRIPTION
**Pull request by fallafeljan**
Referencing #182. Added a small HTTP server (`test/helpers/fixture-server.js`) to perform tests in `test/remote.js`. All passing so far!

Edit: Argh, didn't know each commit shows up individually...

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -166,8 +167,13 @@ function parseStyles(
...
     stmt.media = joinMedia(media, stmt.media)
 
-    // skip protocol base uri (protocol://url) or protocol-relative
-    if (stmt.type !== "import" || /^(?:[a-z]+:)?\/\//i.test(stmt.uri)) {
+    if (isRemote(stmt.uri)) {
+      if (stmt.uri.indexOf("//") === 0) {
+        stmt.uri = "http:" + stmt.uri
+      }
+    }
+    // skip non-html protocol base uri (protocol://url)
```
Comment (TrySound) [inline index.js]: non-**http**

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
+# 8.0.3 - 2015-03-06
```
Comment (TrySound) [inline CHANGELOG.md]: Remove header

File: index.js [hunk 3]
```diff
@@ -221,8 +227,14 @@ function resolveImportId(
...
 
-  return Promise.resolve(options.resolve(stmt.uri, base, options))
-  .then(function(resolved) {
+  return new Promise(function(resolve) {
+    if (isRemote(stmt.uri)) {
+      resolve(stmt.uri)
+    }
+    else {
+      resolve(options.resolve(stmt.uri, base, options))
+    }
```
Comment (TrySound) [inline index.js]: Better use ternary operator and pass result in `Promise.resolve()`

**Comment by TrySound** [comment]
@fallafeljan Can you make this feature inside resolver and loader without editing `index.js`?

File: package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "postcss-import",
-  "version": "8.0.2",
+  "version": "8.0.3",
```
Comment (TrySound) [inline package.json]: Don't do this too,

**Comment by fallafeljan** [comment]
Not without editing index.js at all, since all protocol-based imports are ignored by default. I'm looking into putting most feature code into resolver & loader and just change the protocol matching regex to pass http://, https:// and //.

**Comment by TrySound** [comment]
@fallafeljan Okay. We can not filter url in the next major release.

**Comment by TrySound** [comment]
@fallafeljan Can you choose something different of `request-promise`. That dep seems too big.

**Comment by fallafeljan** [comment]
@TrySound sure, I'll just look for something with less deps – any recommendations?

Pardon, what do you mean by "filter url"? Changing the index.js protocol filtering?

**Comment by TrySound** [comment]
@fallafeljan We can pass false in resolve function or maybe add some fitler option.

**Comment by fallafeljan** [comment]
So moving the protocol checking from `parseStyles` of `index.js` into `resolve-id.js` and filter by rejecting the promise in `resolveId`?

**Comment by TrySound** [comment]
@fallafeljan Not now. When we will be reade for major. Now just leave it.

File: CHANGELOG.md (hunk 2 above)
Comment (MoOx) [inline CHANGELOG.md]: ping

File: package.json (hunk 4 above)
Comment (MoOx) [inline package.json]: +1

**Comment by MoOx** [comment]
@TrySound I found your comment confusing.

It's not a big deal to make a breaking change.

**Comment by TrySound** [comment]
@MoOx Yeah, I know. I just don't have a lot time on my new work for open source these days. Maybe next week.

**Comment by MoOx** [comment]
Ok no hurry, I just wanted to know if you were ok ;)

**Comment by ElRaph** [comment]
If someone else needs this functionality, it could be found there: https://github.com/unlight/postcss-import-url

**Comment by MoOx** [comment]
Cool! So I am closing!

**Pull request closed without merging**